pgjdbc/src/main/java/org/postgresql/jdbc/PgClob.java

```diff
@@ -20,60 +20,110 @@ public PgClob(org.postgresql.core.BaseConnection conn, long oid) throws java.sql
     super(conn, oid);
   }
 
-  public synchronized Reader getCharacterStream(long pos, long length) throws SQLException {
-    checkFreed();
-    throw org.postgresql.Driver.notImplemented(this.getClass(), "getCharacterStream(long, long)");
+  public Reader getCharacterStream(long pos, long length) throws SQLException {
+    lock.lock();
```

Comment:
is this even needed? does checkFreed use any necessary syncrhonization for its behavior?